### PR TITLE
test: refactor header inclusion to speed up building (for test/common/http)

### PR DESCRIPTION
**Pull request by foreseeable**
Signed-off-by: Muge Chen <mugechen@google.com>

Commit Message: refactor header inclusion to speed up building
Additional Description: This is the follow up PR for dividing the monolithic mock header test/mocks/upstream/mocks.h #12048
We refactored mock class include directives, this will reduce the target size and building time.
Risk Level: low
Testing: existing tests
Docs Changes: N/A
Release Notes: no

### CODE REVIEW

**Comment by derekargueta** [comment]
Out of curiosity, do you have benchmark numbers of how much this speeds up the build?

**Comment by foreseeable** [comment]
> Out of curiosity, do you have benchmark numbers of how much this speeds up the build?

@derekargueta 
Here is the building time comparison for affected tests:

For some test the building time does not change much, since `upstream_mocks` is not the bottleneck in compilation.

|test_name|after|before|
|---|---|---|
|//test/common/http:async_client_impl_test|91.241s| 113.855s |
|//test/common/http:codec_client_test|72.036s|71.402s|
|//test/common/http:conn_manager_impl_fuzz_test| 81.805s |95.341s|
|//test/common/http:conn_manager_impl_test|106.559s|106.681s|
|//test/common/http/http1:conn_pool_test|70.243s|72.604s|
|//test/common/http/http2:conn_pool_test|73.546s|67.012s|

**Comment by derekargueta** [comment]
Super cool, thanks for sharing!

File: test/common/http/http1/BUILD [hunk 1]
```diff
@@ -60,7 +60,8 @@ envoy_cc_test(
         "//test/mocks/http:http_mocks",
         "//test/mocks/network:network_mocks",
         "//test/mocks/runtime:runtime_mocks",
-        "//test/mocks/upstream:upstream_mocks",
+        "//test/mocks/upstream:cluster_info_mocks",
+        "//test/mocks/upstream:transport_socket_match_mocks",
```
Comment (asraa) [inline test/common/http/http1/BUILD:64]: was this a transitive or missing dep?

Comment (foreseeable) [reply test/common/http/http1/BUILD:64]: the `transport_socket_match_mocks` target was a missed dep and I added it into rules

**Comment by rgs1** [comment]
@foreseeable was this done with a tool or manually?

**Comment by foreseeable** [comment]
it's done with a tool @rgs1 #12150

**Pull request merged**